### PR TITLE
KIALI-1284 Update Health to use Workloads instead Deployments

### PR DESCRIPTION
**Pull request by lucasponce**
** Describe the change **

UI side of https://github.com/kiali/kiali/pull/500

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#82](https://paste.fedoraproject.org/paste/19Ws3NiHNOYlwPWTmRNJXg)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1317]](https://paste.fedoraproject.org/paste/8AXRmsRMXyc4FEe4DHcfJg)
  * :heavy_check_mark: Istio-Upstream [#[199]](https://paste.fedoraproject.org/paste/QqwSrtVwpsxKIHYBoFXTug)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1160, 1161]](https://paste.fedoraproject.org/paste/XI59FPwN9iQXnwJDUhs4Pw)
  * :heavy_check_mark: install-bookinfo [#[668]](https://paste.fedoraproject.org/paste/mPZebC6jprv1k1fo6o-qKQ)
  * :heavy_check_mark: Kiali-on-Openshift [#[684]](https://paste.fedoraproject.org/paste/5ciH0uNV6l-UdXV8fkW0UQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[132]](https://paste.fedoraproject.org/paste/EIuS~sNqCtlrwB~xA7yWWw)


KIALI_NAME=kialiuipr687-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#104](https://paste.fedoraproject.org/paste/vGGpQOzK-BzhLFlmxBPbhw)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1375]](https://paste.fedoraproject.org/paste/VmGAaXGhTBXvM3kqiePHzg)
  * :heavy_check_mark: Istio-Upstream [#[251]](https://paste.fedoraproject.org/paste/M9IQhjeVYtzpArWQgQLUPA)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1296, 1298]](https://paste.fedoraproject.org/paste/Aeba9UhvSC~2Ml7WdVZixA)
  * :heavy_check_mark: install-bookinfo [#[734]](https://paste.fedoraproject.org/paste/rf-UYGmFtJgpPgVaq1xCVA)
  * :heavy_check_mark: Kiali-on-Openshift [#[745]](https://paste.fedoraproject.org/paste/GHBBByDq~jTUR8SmyEZl-g)
  * :heavy_check_mark: run-kiali-e2e-tests [#[192]](https://paste.fedoraproject.org/paste/SXvk0mjonv73ziRpNRK1gw)


KIALI_NAME=kialiuipr687-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#121](https://paste.fedoraproject.org/paste/s0Fn8rwURwFwKjWczwxwIg)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1405]](https://paste.fedoraproject.org/paste/MjehAzprYXvw6oGKtcb8Ug)
  * :red_circle: Istio-Upstream [#[278]](https://paste.fedoraproject.org/paste/9XPapr-0uHDEKVt-cIV81w)


KIALI_NAME=kialiuipr687-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#122](https://paste.fedoraproject.org/paste/XdtYNKAFiiT9Z5~zJRZpVw)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1406]](https://paste.fedoraproject.org/paste/7AZaxATL04pe5fRItli~AQ)
  * :heavy_check_mark: Istio-Upstream [#[279]](https://paste.fedoraproject.org/paste/ibqB9m9yrJY2OT2QxXyuEw)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1361, 1362]](https://paste.fedoraproject.org/paste/KckJ-xSvVixL-xMjK6ZQCQ)
  * :heavy_check_mark: install-bookinfo [#[771]](https://paste.fedoraproject.org/paste/DPqpICFZ1xjiDjUoTvFZuw)
  * :heavy_check_mark: Kiali-on-Openshift [#[774]](https://paste.fedoraproject.org/paste/djnE1UR80P7YUUdeoQkcwA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[221]](https://paste.fedoraproject.org/paste/rS5IGB0BsvXDuIPVJOJd5g)


KIALI_NAME=kialiuipr687-istio-system

File: src/types/Health.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export interface EnvoyRatio {
   total: number;
 }
 
-export interface DeploymentStatus {
+export interface WorkloadStatus {
```
Comment (jotak) [inline src/types/Health.ts]: I wonder if "ReplicasStatus" would be a more accurate name, after all. This health component aims to show how many replcias are running versus how many are desired. What do you think?

**Comment by hhovsepy** [comment]
List workloads.
Apply filter by namespace.
Clear filters.
Go to Apps or any other page.
Back to Workloads.
Empty list.

**Comment by lucasponce** [comment]
Thanks for testing @hhovsepy.
It sounds I missed something in the backend for deployments.
I will check.
Expected workload list is something like this:
image

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#152](https://paste.fedoraproject.org/paste/ARtn59Lk~k7wmdiwCFT6ew)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1469]](https://paste.fedoraproject.org/paste/iKxJGSoZsQ9YNwAh~XiIWg)
  * :heavy_check_mark: Istio-Upstream [#[335]](https://paste.fedoraproject.org/paste/fZnUFj1HM1ay9oh8iMgkFg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1488, 1487]](https://paste.fedoraproject.org/paste/3ao0y5ws3j4UTBd0w3Yslg)
  * :heavy_check_mark: install-bookinfo [#[839]](https://paste.fedoraproject.org/paste/659Qbmf1ZO5wiwaw3vh57Q)
  * :heavy_check_mark: Kiali-on-Openshift [#[839]](https://paste.fedoraproject.org/paste/Rott8IM9ZWASTp0RdhhNUA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[272]](https://paste.fedoraproject.org/paste/v33n7TneBpetY3xxb3gATw)


KIALI_NAME=kialiuipr687-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#153](https://paste.fedoraproject.org/paste/FUe4NL1KgBIROnN5pJV-Jw)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1471]](https://paste.fedoraproject.org/paste/QBxDJpGlLPrAN5PiKnSNIg)
  * :heavy_check_mark: Istio-Upstream [#[337]](https://paste.fedoraproject.org/paste/ztdRsT47hB6xxenEe4fQFg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1491, 1492]](https://paste.fedoraproject.org/paste/9AfA0Cw4rdH9KyhohJfTrQ)
  * :heavy_check_mark: install-bookinfo [#[841]](https://paste.fedoraproject.org/paste/FDT7vf7ctsmYiITAKc8iuA)
  * :heavy_check_mark: Kiali-on-Openshift [#[841]](https://paste.fedoraproject.org/paste/gz1AD~4KToIE29d8INmhVg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[274]](https://paste.fedoraproject.org/paste/SkT5w91puzQTfhWmaEji2g)


KIALI_NAME=kialiuipr687-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#155](https://paste.fedoraproject.org/paste/FINLry0KfBcuoYFkPcZRtQ)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1475]](https://paste.fedoraproject.org/paste/cSkw0P2QF-xP6ROpqWgqsw)
  * :heavy_check_mark: Istio-Upstream [#[340]](https://paste.fedoraproject.org/paste/tvMhwSF950qzmYyvBmZuPw)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1499, 1500]](https://paste.fedoraproject.org/paste/8rjUV64EqhIVKaYGyvzGbA)
  * :heavy_check_mark: install-bookinfo [#[845]](https://paste.fedoraproject.org/paste/lRSE04lq7dqnRrdu--wgAQ)
  * :heavy_check_mark: Kiali-on-Openshift [#[845]](https://paste.fedoraproject.org/paste/LizDbPbV5x9lupuEX6xlFA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[278]](https://paste.fedoraproject.org/paste/ieQE~RdkwSpt0FWQhdDqmA)


KIALI_NAME=kialiuipr687-istio-system

File: src/types/Health.ts (hunk 1 above)
Comment (lucasponce) [reply src/types/Health.ts]: I remember you commented same thing in the backend PR and we finish to leave as WorkloadStatus. I would maintain backend -> frontend mapping for consistency, too.

**Comment by lucasponce** [comment]
@hhovsepy @jotak this is related to PR 500 from backend.
Just let me know if you see anything suspicious.
Thanks.

**Comment by lucasponce** [comment]
@hhovsepy @jotak as Workload Type could be a long list I am thinking to use the free text to not select between controller type, but let to type it.

**Comment by hhovsepy** [comment]
> @hhovsepy @jotak as Workload Type could be a long list I am thinking to use the free text to not select between controller type, but let to type it.
Then we need a documentation Jira there to describe all types.
btw, case sensitive or not?

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#178](https://paste.fedoraproject.org/paste/BtGqn9C9yVmt1Er0klZm4g)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1523]](https://paste.fedoraproject.org/paste/1KwtYCTZGCOBeQ~3InO1Ng)
  * :heavy_check_mark: Istio-Upstream [#[381]](https://paste.fedoraproject.org/paste/C5RxLp8qsBbb-baKtb42Hg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1596, 1597]](https://paste.fedoraproject.org/paste/2VwD6XYwJ~7UlqdDhULIzQ)
  * :heavy_check_mark: install-bookinfo [#[893]](https://paste.fedoraproject.org/paste/mBS4br99n8d9LqTkRY9-kQ)
  * :heavy_check_mark: Kiali-on-Openshift [#[896]](https://paste.fedoraproject.org/paste/zJlBXuR40zXseE7alZAcHg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[321]](https://paste.fedoraproject.org/paste/qu9aHv6bewLazE1J8ZZfrg)


KIALI_NAME=kialiuipr687-istio-system

**Comment by hhovsepy** [comment]
@lucasponce the Workload Type filter does not have options yet, right?

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#187](https://paste.fedoraproject.org/paste/maVwWeqa3ofqGwB4fkcy-w)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1544]](https://paste.fedoraproject.org/paste/4Ik24zY86JB4wt5zKpu3iA)
  * :heavy_check_mark: Istio-Upstream [#[401]](https://paste.fedoraproject.org/paste/rxuwGqh44uxl34n92ljFbA)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1641, 1643]](https://paste.fedoraproject.org/paste/YrmtTsomG-QwL9rRkVcFhA)
  * :heavy_check_mark: install-bookinfo [#[914]](https://paste.fedoraproject.org/paste/c5rg7TAnPpKt~GAOutzTEQ)
  * :heavy_check_mark: Kiali-on-Openshift [#[921]](https://paste.fedoraproject.org/paste/OK-1-XhO02hnCC4aDenUrw)
  * :heavy_check_mark: run-kiali-e2e-tests [#[342]](https://paste.fedoraproject.org/paste/taHyKqmFm3qQmfe-9QbfHQ)


KIALI_NAME=kialiuipr687-istio-system

**Comment by lucasponce** [comment]
> @lucasponce CronJobs are not listed, only Jobs

Right, on this case Jobs are infered from Pods, but CronJob is missed.
So, I think we need to add it as it can flood easily the workload list.

File: src/pages/WorkloadDetails/WorkloadDetailsPage.tsx [hunk 2]
```diff
@@ -53,17 +38,48 @@ class WorkloadDetails extends React.Component<RouteComponentProps<WorkloadId>, W
...
+          checks: []
+        };
+        if (!pod.istioContainers || pod.istioContainers.length === 0) {
+          validations['pod'][pod.name].checks.push(noIstiosidecar);
+        }
+        if (!pod.labels) {
+          validations['pod'][pod.name].checks.push(noAppLabel);
+          validations['pod'][pod.name].checks.push(noVersionLabel);
+        } else {
+          if (!('app' in pod.labels)) {
```
Comment (jotak) [inline src/pages/WorkloadDetails/WorkloadDetailsPage.tsx]: these label names 'app' and 'version' should not be hardcoded
They could be passed in the workloads results, or another possibility would be to perform labels validation from the backend only.

Comment (lucasponce) [reply src/pages/WorkloadDetails/WorkloadDetailsPage.tsx]: +1
Pod validation can be done from the model.Pod as it is created.
I willl address that.
Thanks for checking it.

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#193](https://paste.fedoraproject.org/paste/jKD-V-XBNcQWyAJlaY94yQ)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1558]](https://paste.fedoraproject.org/paste/lGIdWqhnqEYIs4IPJLOnTw)
  * :heavy_check_mark: Istio-Upstream [#[414]](https://paste.fedoraproject.org/paste/16uxL8KbIccNSzpk~1d7qg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1672, 1670]](https://paste.fedoraproject.org/paste/bCdEYkqQlmfa2lgSTUJ-tg)
  * :heavy_check_mark: install-bookinfo [#[927]](https://paste.fedoraproject.org/paste/aj6rpsgFsuTTtFylFFSdSg)
  * :heavy_check_mark: Kiali-on-Openshift [#[936]](https://paste.fedoraproject.org/paste/Y729kXcTxtp9BdQuVoEbzg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[358]](https://paste.fedoraproject.org/paste/wTAcHRBZLX2CYrSKJlqW-w)


KIALI_NAME=kialiuipr687-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#199](https://paste.fedoraproject.org/paste/Y6BZQaivEmoJIl5qguXKVw)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1570]](https://paste.fedoraproject.org/paste/0DgP2TF4~bet8JGUVmpNGQ)
  * :heavy_check_mark: Istio-Upstream [#[424]](https://paste.fedoraproject.org/paste/wthY4Gk~lM~elfsvETqqzw)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1698, 1700]](https://paste.fedoraproject.org/paste/n4efzKg2R3VS~CJI90YFQg)
  * :heavy_check_mark: install-bookinfo [#[942]](https://paste.fedoraproject.org/paste/~j9euqi4ADf8sQJddBUqNA)
  * :heavy_check_mark: Kiali-on-Openshift [#[953]](https://paste.fedoraproject.org/paste/3MAdap4GEH1D5aUbz5ZTMA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[368]](https://paste.fedoraproject.org/paste/lTzzdgJp5LIuVPmJ0tZYLA)


KIALI_NAME=kialiuipr687-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#200](https://paste.fedoraproject.org/paste/YdVx1mII6HER6FpV2gfBuA)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1572]](https://paste.fedoraproject.org/paste/W4HKYL4GT9SwQ-ivCm8QFA)
  * :heavy_check_mark: Istio-Upstream [#[426]](https://paste.fedoraproject.org/paste/7ic8v6fzbici2CUSuGfkug)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1704, 1703]](https://paste.fedoraproject.org/paste/xxKZ2rHHldlNLQ-OWtACrg)
  * :heavy_check_mark: install-bookinfo [#[944]](https://paste.fedoraproject.org/paste/pyH9SF8-wnQv7~l3uRE0Ww)
  * :heavy_check_mark: Kiali-on-Openshift [#[955]](https://paste.fedoraproject.org/paste/0j51j6fon97D0rWcC44eYQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[370]](https://paste.fedoraproject.org/paste/mgPmlPSEM3XbF8tSd0ifng)


KIALI_NAME=kialiuipr687-istio-system

**Comment by hhovsepy** [comment]
Verified

**Pull request merged**